### PR TITLE
Remove double soft-max in load-balancing loss. Fixes #39055 .

### LINKED ISSUE #39055: DBRX model passes probabilities and not logits to the load balancer
### System Info

The problem has nothing to do with my system.

### Who can help?

@ArthurZucker

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Look into the code and verify that it is not compliant with the [intended load balancing](https://arxiv.org/pdf/2101.03961):

The signature of the load balancer in [modeling_dbrx](https://github.com/huggingface/transformers/blob/main/src/transformers/models/dbrx/modeling_dbrx.py) looks as follows:

```python
def load_balancing_loss_func(
    gate_logits: torch.Tensor,
    num_experts: int,
    top_k: int,
    attention_mask: Optional[torch.Tensor],
) -> torch.Tensor:
```

And also, the `gate_logits` are treated as such, meaning that the softmax is computed later in the code. However, the values (called `weights`) that are passed into that function are the expert probabilities and not the logits:

```python
weights = self.layer(hidden_states).softmax(dim=-1, dtype=torch.float32)
```

Thus, the load balancer takes the softmax of the logits twice, which is incorrect.

### Expected behavior

`gate_logits` should be logits and not probabilities.

### PR DESCRIPTION
**Pull request by rudolfwilliam**
# What does this PR do?

Fixes #39055 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
3-4 other models are affected by this it seems? Could you make sure they get the fix as well? (qwen moe, jetmoe etc)

**Comment by rudolfwilliam** [comment]
I went trough some other classes (but not all) that use the same load balancer, like qwen moe and jetmoe. It seems they implement load balancing correctly (they either operate explicitly on the probabilities-as implemented in my pull request- or correctly pass logits to the function).

**Review by ArthurZucker — APPROVED** [review]
Thanks for checking
